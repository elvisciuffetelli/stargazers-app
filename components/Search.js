import React from "react";
import { View } from "react-native";
import { Button, Input, Text } from "@rneui/themed";
import PropTypes from "prop-types";
import styles from "../styles";

export default function Search({
  handlePress,
  onChangeOwner,
  owner,
  onChangeRepo,
  repo,
  fieldError,
  errorMessage,
}) {
  return (
    <View style={styles.inputContainer}>
      <Input
        style={styles.input}
        onChangeText={onChangeOwner}
        value={owner}
        placeholder="Repo owner"
        errorStyle={{ color: "firebrick" }}
        errorMessage={fieldError}
      />
      <Input
        style={styles.input}
        onChangeText={onChangeRepo}
        value={repo}
        placeholder="Repo name"
        errorStyle={{ color: "firebrick" }}
        errorMessage={fieldError}
      />
      {errorMessage && <Text style={styles.errorMessage} h6>{errorMessage}</Text>}
      <Button
        title="SEARCH"
        size="md"
        onPress={handlePress}
        buttonStyle={styles.button}
      />
    </View>
  );
}

Search.defaultProps = {
  owner: "",
  repo: "",
  fieldError: "",
  errorMessage: "",
};

Search.propTypes = {
  handlePress: PropTypes.func.isRequired,
  onChangeOwner: PropTypes.func.isRequired,
  onChangeRepo: PropTypes.func.isRequired,
  owner: PropTypes.string,
  repo: PropTypes.string,
  fieldError: PropTypes.string,
  errorMessage: PropTypes.string,
};
