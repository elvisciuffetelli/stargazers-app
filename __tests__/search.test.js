import React from "react";
import renderer from "react-test-renderer";

import Search from "../components/Search";

describe("<Grid />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot(); // The length of the tree should be three because we want a 3x3 grid
  });
});
