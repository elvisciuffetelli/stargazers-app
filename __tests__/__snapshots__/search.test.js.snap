// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> has 1 child 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
    testID="RNE__Input__view-wrapper"
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        editable={true}
        placeholder="Repo owner"
        placeholderTextColor="#86939e"
        style={
          Object {
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        Object {
          "color": "red",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
    testID="RNE__Input__view-wrapper"
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        editable={true}
        placeholder="Repo name"
        placeholderTextColor="#86939e"
        style={
          Object {
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        Object {
          "color": "red",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 2,
        },
        undefined,
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderColor": "#2089dc",
            "borderRadius": 30,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 20,
            "padding": 8,
            "paddingHorizontal": 10,
            "width": 150,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          SEARCH
        </Text>
      </View>
    </View>
  </View>
</View>
`;
